kv/dist_sender.go

```diff
@@ -396,6 +403,15 @@ func (ds *DistSender) Send(call *client.Call) {
 				}
 			} else if isMulti {
 				// If this request spans ranges, collect the replies.
+				if maxResults > 0 {
+					if reply, ok := reply.(proto.Truncatable); ok {
```

Comment:
Added: Don't let the #494 stuff distract you from the main point of the PR - I just wanted to mention it. Looks like the proper fix for that is actually upstream in txn_coord_sender, so it should probably be a separate PR anyways.